src/main/scala/ignition/core/cache/ExpiringMultiLevelCache.scala

```diff
@@ -370,7 +411,18 @@ case class ExpiringMultiLevelCache[V](ttl: FiniteDuration,
           case None =>
             // There are no remote RW caches
             logger.error(s"canonicalValueGenerator, key $key: failed to generate value and no remote cache configured", eLocal)
-            Future.failed(eLocal)
+            eLocal match {
+              case NonFatal(e) => {
+                if (cacheErrors) {
+                  // if error was NonFatal Error then saves it to cache
+                  val timestampedValue = timestamp(hasError = true, throwable = e)
```

Comment:
I think the throwable is good enough to say it has an error or not.
I'd remove the hasError flag. It's redundant